### ISSUE TITLE
Error: Provider produced inconsistent result after apply

### ISSUE THREAD

**Issue by Uvindu96** [hashicorp/terraform-provider-time#57]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritise this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritise the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform (and AzureAD Provider) Version
- Terraform version v0.14.6
- AzureAD Provider version  2.10.0

### Affected Resource(s)

* `time_rotating`

### Description 
For the AD app secret rotation I have configured a `time_rotating` resource and `rotate_when_changed` attribute to the `azuread_service_principal_password` resource.

```
resource "time_rotating" "test" {
  rotation_days = 20
  lifecycle {
    create_before_destroy = true
  }
}
resource "random_password" "password" {
  length            = var.password_length
  special           = true
  lifecycle {
    create_before_destroy = true
  }
}

resource "azuread_service_principal_password" "service_principal_password" {
  service_principal_id = var.sp_internal_id
   rotate_when_changed = {
    rotation = time_rotating.test.id
  }
  lifecycle {
    create_before_destroy = true
  }
}
````
**According to this configuration I can change the rotation by days, but when I try to set the rotation based on minutes by using the attribute `rotation_minutes` it gives me the following error. And recently when I changed the value for same attribute `rotation_minutes` it gave me the same error.**

```
Error: Provider produced inconsistent result after apply

When applying changes to
module.test-storage-account-dev-ops-ad-sp-password.time_rotating.test,
provider "registry.terraform.io/hashicorp/time" produced an unexpected new
value: Root resource was present, but now absent.

This is a bug in the provider, which should be reported in the provider's own
issue tracker.
```
For the observation here I will attach the log file.
Log File -:
[apply-log.txt](https://github.com/hashicorp/terraform-provider-time/files/7608162/apply-log.txt)
Is there any solution for this error ? Or is there any workaround that I can follow ? much appreciate your help on this :)

**Comment by derekwilliamsliquidx** [comment]
I have discovered that the issue lies when the property `rotation_<days/years/minutes>` changes to something else. This causes the output to change (I assume the format). If you have changed this property recently to something other than what's in your state, revert the change.

**Comment by fuyong2022** [comment]
@derekwilliamsliquidx @Uvindu96  I hit the same issue and it appears to be a bug. Any plan to fix this?

**Comment by fuyong2022** [comment]
One workaround is to manually remove the resource through terraform command, then it will force time resource to be re-created.

**Comment by rdvlvtflr** [comment]
> 

Hi @fuyong2022 when you state "manually remove the resource" - to which resource are you referring to?

**Comment by SBGoods** [comment]
Hi @Uvindu96, thank you for raising this issue and we apologize for the delay. We recently migrated the Time provider from Terraform Plugin SDKv2 to Terraform Plugin Framework and solved a similar issue #133 with the `time_rotating` resource that was released in v0.9.1. Can you upgrade the Time Provider to v0.9.1 and see if you can replicate the same issue in that version? 

Please reach out if you have any further questions!

**Comment by SBGoods** [comment]
Hi @Uvindu96, since we haven't heard back from you, I will close this issue for now. If you are still running into trouble with this, please feel free to reach out and open another issue.
